### DISCUSSION TITLE
about how can i start a micropython transplantation to new chips?

### DISCUSSION THREAD

**Discussion by QiaoTuCodes** [micropython/micropython#10853] [Core Development] [answered]
hey guys. i'm serious that. i want know about how start a micropython transplantation to new chips? i mean create a new ports. what are the steps and development processes? please give me some directions and suggest. thank you so much for everyone.!

**Comment by mattytrentini** [comment, marked as answer]
The first place to start would be the [Porting MicroPython](https://docs.micropython.org/en/latest/develop/porting.html) doc. It provides an overview of the concepts as well as some basic examples. The [bare-arm port](https://github.com/micropython/micropython/tree/master/ports/bare-arm) is also useful to look at since is a small example of an actual port working - albeit with very few features.

From there you'll probably need to refer to other ports to see how things are done. I often first turn to the [rp2 port](https://github.com/micropython/micropython/tree/master/ports/rp2/) since it's one of the most recent and - especially since there's only one micro in the port - the code is quite clean. The only warning is that the build system can be confusing on that port...

**Reply by QiaoTuCodes** [reply]
okey,thank you so much. i'll take a look right now.
